src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netcore.cs

```diff
@@ -2638,42 +2655,43 @@ private static float MinMaxCore<TMinMaxOperator>(ReadOnlySpan<float> x)
                 if (i != x.Length)
                 {
                     current = Vector128.LoadUnsafe(ref xRef, (uint)(x.Length - Vector128<float>.Count));
-                    if (!Vector128.EqualsAll(current, current))
+
+                    nanMask = Vector128.Equals(current, current);
+                    if (nanMask != Vector128<float>.AllBitsSet)
                     {
-                        return GetFirstNaN(current);
+                        return current.GetElement(IndexOfFirstMatch(nanMask));
                     }
 
-                    result = Vector128.ConditionalSelect(
-                        Vector128.Equals(CreateRemainderMaskSingleVector128(x.Length - i), Vector128<float>.Zero),
+                    result = ElementWiseSelect(
+                        CreateRemainderMaskSingleVector128(x.Length - i),
                         result,
-                        TMinMaxOperator.Invoke(result, current));
+                        TMinMaxOperator.Invoke(result, current)
+                    );
                 }
 
                 // Aggregate the lanes in the vector to create the final scalar result.
                 return TMinMaxOperator.Invoke(result);
             }
 
             // Scalar path used when either vectorization is not supported or the input is too small to vectorize.
+            float curResult = x[0];
```

Comment:
Why switch from result to curResult? Keeping the naming here consistent with the naming used in the vectorized blocks seemed worthwhile enough to make needing the extra scoping worthwhile.

Reply:
Matches how we were doing it in the `IndexOf*` methods and made it simpler to compare/test the two.

Longer term this will get rewritten when we add alignment support and can be fixed up then to follow the pattern we have in our other functions, where it's separated out.